db/data_migration/20230104104917_reslug_her_majesty_to_his_majesty.rb

```diff
@@ -0,0 +1,105 @@
+require "csv"
+require "uri"
+
+csv_file = File.join(File.dirname(__FILE__), "20230104104916_reslug_her_majesty_to_his_majesty.csv")
+csv = CSV.parse(File.open(csv_file), headers: true)
+
+csv.each do |row|
+  puts "reslugging #{row['current_url']}"
+
+  old_slug = URI(row["current_url"]).path.split("/").last
+  new_slug = URI(row["new_url"]).path.split("/").last
+
+  case row["content_type"]
+  when "html_publication"
+    publication_slug = URI(row["current_url"]).path.split("/").second_to_last
+    documents = Document.where(slug: publication_slug)
+    if documents.count > 1
+      puts "There are multiple documents with the slug '#{publication_slug}'. Consider writing a migration and fetching the document with a content_id or document_type to uniquely identify it."
+      next
+    end
+
+    document = documents.first
+    edition = document&.editions&.published&.last
+    unless edition
+      puts "!! edition not found for: #{old_slug}"
+      next
+    end
+
+    html_attachment = edition.attachments.find_by(slug: old_slug)
+
+    unless html_attachment
+      puts "!! no HTML attachments found for: #{old_slug}"
+      next
+    end
+
+    # update slug of html attachement and add redirect
+    html_attachment.update!(slug: new_slug)
+    Whitehall::PublishingApi.republish_async(html_attachment)
+
+    # remove the most recent edition from the search index
+    Whitehall::SearchIndex.delete(edition)
+
+    # send edition to publishing api
+    PublishingApiDocumentRepublishingWorker.new.perform(document.id)
+
+    # add edition to search index
+    Whitehall::SearchIndex.add(edition)
```

Comment:
Also: could we use the `DataHygiene::DocumentReslugger` to handle this – once we've located the HTML attachment's document and updated the slug?

(We think probably not, because this is to update the slug of individual HTML attachments rather than the parent publication.)

If not: it would be good to follow-up by creating a `HtmlAttachmentReslugger` for future re-use.